### ISSUE TITLE
train! doesn't work with Trackedreal

### ISSUE THREAD

**Issue by merckxiaan** [FluxML/Flux.jl#630]
A model with a Trackedreal parameter crashes during the backward pass with the `train!` function
`type TrackedReal has no field grad`

**Comment by alhirzel** [comment]
I believe I am seeing this too, here is one example:

```julia
using Flux: mse, param, params
using Flux.Tracker: update!, TrackedReal
using Flux.Optimise
using Random: seed!
using Plots

struct LinRegParams
	slope     :: TrackedReal{Float64}
	intercept :: TrackedReal{Float64}
end

LinReg(m, b) = LinRegParams(param(m), param(b)) # instantiator
(m::LinRegParams)(x) = m.slope*x .+ m.intercept # evaulator

m = LinReg(1, 0) # m is callable and has attached data

loss(x, y) = mse(m(x), y)
ps = params(m.slope, m.intercept)
data = [(0,3), (1,5)]  # should get us to m=2, b=3

train!(loss, ps, data, Descent(0.1))
@show m(0)
```

**Comment by alhirzel** [comment]
I think this will be addressed when #562 is merged.

**Comment by kota7** [comment]
This is also related, perhaps. https://github.com/FluxML/Flux.jl/issues/214
